### DISCUSSION TITLE
Add ability to fork a database by block number

### DISCUSSION THREAD

**Discussion by roninjin10** [ponder-sh/ponder#148] [Idea / Feature Request]
Building e2e tests for optimism I have realized what would make it really easy is if I could not only fork anvil at a specific block height but also fork my live database at that block height.   Something like this https://neon.tech/branching

**Comment by typedarray** [comment]
Subgraphs support something similar via [time-travel queries](https://github.com/graphprotocol/graph-node/issues/813). Ponder could certainly add a similar feature in the future (note: ponder's "time" would likely need to be timestamp rather than a block number, because ponder supports multiple chains in the same app).

**Reply by roninjin10** [reply]
Even if it is ultimately picking a number the api can still take a block height or a timestamp and just use that block/chain timestamp under the hood
